routes/transactionBreakdown.ts

```diff
@@ -0,0 +1,110 @@
+import { Request, Response } from 'express';
+import express from 'express';
+const router = express.Router();
+
+const createItemList = (items: Array<any>) => {
+  return items.map((individual: any, ind: number) => {
+    const indItem = {
+      id: ind,
+      name: individual.name,
+      people: individual.people,
+      cost: individual.cost,
+    };
+    return indItem;
+  });
+};
+
+const getSubAmount = (orderItems: Array<any>, name: string = '') => {
+  let total = 0;
+  for (const item of orderItems) {
+    if (name) {
+      if (item.people.includes(name)) {
+        total += Math.round((item.cost / item.people.length) * 100) / 100;
+      }
+    } else {
+      total += item.cost;
+    }
+  }
+  return total;
+};
+
+const createOrderTransaction = (
+  items: Array<any>,
+  tax: number,
+  tip: number,
+  party: Array<string>
+) => {
+  const subAmount = getSubAmount(items);
```

Comment:
I also think we should separate the logic for getting subAmount for orderTransaction from the logic for getting subAmount for personCost.

Reply:
Sayy less, i was thinking that too but I think work got me in the 'how can we write this in the least code possible' mindset lol